src/Lecture3.hs

```diff
@@ -89,8 +95,13 @@ weekday to the second.
 >>> daysTo Friday Wednesday
 5
 -}
-daysTo = error "TODO"
 
+daysTo :: Weekday -> Weekday -> Int
+daysTo from to
+  | from > to = delta + fromEnum (maxBound `asTypeOf` from) + 1
```

Comment:
That's an interesting usage of `asTypeOf` 😅 You don't need a function here and you can simply use a type annotation:
```suggestion
  | from > to = delta + fromEnum (maxBound :: Weekday) + 1
```